parse/parser.go

```diff
@@ -514,14 +520,7 @@ func parseFunctionLiteral(buf TokenBuffer) (*ast.FunctionLiteral, error) {
 		return nil, err
 	}
 
-	// After exit from parseBlockStatement, consume right-brace
-	if err = expectNext(buf, Rbrace); err != nil {
```

Comment:
Left brace is parsed in function parsing phase, but Right brace is parsed in block parsing phase.
I think braces parsing should be done in one phase.